### PR TITLE
Clone memref.global ops into GPU modules in convert-rock-to-gpu

### PR DESCRIPTION
**Pull request by krzysz00**
Fixes
https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/764

### CODE REVIEW

File: mlir/lib/Conversion/RockToGPU/RockToGPU.cpp [hunk 1]
```diff
@@ -181,6 +184,24 @@ void LowerRockOpsToGPUPass::runOnOperation() {
...
+          SymbolRefAttr globalSym = op.getNameAttr();
+          auto toClone = dyn_cast_or_null<memref::GlobalOp>(
+              SymbolTable::lookupNearestSymbolFrom(op, globalSym));
+          if (!toClone)
+            return WalkResult::interrupt();
+          OpBuilder::InsertionGuard guard(b);
+          Operation *cloned = toClone.clone();
+          // There probably shouldn't be any renames, but let's be careful.
+          StringAttr newNameAttr = gpuModuleSymbolTable.insert(cloned);
+          op.setNameAttr(FlatSymbolRefAttr::get(newNameAttr));
```
Comment (sjw36) [inline mlir/lib/Conversion/RockToGPU/RockToGPU.cpp]: But now you will have to update references if the name changed. Probably best to assert that it didn't change.

**Review by sjw36 — COMMENTED** [review]
Just 1 caveat.

File: mlir/lib/Conversion/RockToGPU/RockToGPU.cpp (hunk 1 above)
Comment (krzysz00) [reply mlir/lib/Conversion/RockToGPU/RockToGPU.cpp]: Ok, I've now tried to do something about that (and the possibility that we might clone the same constant twice)

**Review by sjw36 — APPROVED** [review]
LGTM.

**Pull request merged**